dcicutils/misc_utils.py

```diff
@@ -1630,16 +1631,140 @@ class SubFreeze(Freeze):
+                if isinstance(attribute_value, cls):
+                    return superclass, attribute_value
+                raise ValueError(f"The slot {instance_class.__name__}.{attribute_name}"
+                                 f" does not contain a cached value.")
+        raise ValueError(f"The slot {instance_class.__name__}.{attribute_name} is not defined.")
+
+    @classmethod
+    def _find_cache_class(cls, instance_class, attribute_name):
+        return cls._find_cache_class_and_attribute(instance_class, attribute_name)[0]
+
+    @classmethod
+    def _find_cache_attribute(cls, instance_class, attribute_name):
+        return cls._find_cache_class_and_attribute(instance_class, attribute_name)[1]
+
+    @classmethod
+    def _find_cache_map(cls, instance_class, attribute_name) -> dict:
+        attribute = cls._find_cache_attribute(instance_class, attribute_name)
+        return attribute.attribute_cache
+
+    @classmethod
+    def reset(cls, *, instance_class, attribute_name, subclasses=True):
+        """
+        Clears the cache for the given attribute name on the given instance_class.
+
+        Having done:
+
+            import random
+            class Foo:
+                @classproperty_cached
+                def something(cls):
+                    return random.randint(100)
+
+        one can clear the cache by doing:
+
+            classproperty_cached.reset(instance_class=Foo, attribute_name='something')
+
+        :param instance_class: a class with an attribute whose value is managed by '@classproperty_cached'.
+        :param attribute_name: the name of the attribute that has a cached value.
+        :param subclasses: a bool indicating whether the cache reset requests applies to
+             all subclasses (subclasses=True) or only the exact class given (subclasses=False).
+             Using subclasses=False is not allowed if CACHE_EACH_SUBCLASS is False.
+        """
+
+        if not cls._USES_PER_SUBCLASS_CACHES and not subclasses:
+            raise ValueError(f"The subclasses= argument to {cls.__name__}.reset must not be False"
+                             f" because {cls.__name__} does not use per-subclass caches.")
+        if not isinstance(instance_class, type):
+            raise ValueError(f"The instance_class= argument to {cls.__name__}.reset must be a class.")
+        reference_class = cls._find_reference_class(instance_class, attribute_name)
+        attribute_cache = cls._find_cache_map(reference_class, attribute_name)
+        keys_to_remove = []
+        predicate = issubclass if subclasses else equals
+        for cache_class, cache_value in attribute_cache.items():
+            if predicate(cache_class, reference_class):
+                keys_to_remove.append(cache_class)
+        for cache_class in keys_to_remove:
+            del attribute_cache[cache_class]
+        return bool(keys_to_remove)
+
+
+class classproperty_cached_each_subclass(classproperty_cached):
```

Comment:
Could you explain why this is named this way? Why not `classproperty_cached_first_use`?

Reply:
If you have several subclasses of the indicated class, each will have its own value computed. None will share.